lib/route/src/Obelisk/Route.hs

```diff
@@ -312,12 +315,53 @@ isoEncoder f = unsafeMkEncoder $ EncoderImpl
   , _encoderImpl_decode = pure . view (from f)
   }
 
+-- | Lift an 'Iso' over the decoded format to an 'Iso' over encoders
+isoViaDecoded
+  :: (Functor check, Functor parse)
+  => Iso' d d'
+  -> Iso' (Encoder check parse d e) (Encoder check parse d' e)
+isoViaDecoded i = withIso isoImpl $ \forward backward ->
+  iso
+    (Encoder . fmap forward . unEncoder)
+    (Encoder . fmap backward . unEncoder)
+  where
+    isoImpl = withIso i $ \forward backward -> iso (f backward forward) (f forward backward)
+    f bw fw impl = EncoderImpl
+      { _encoderImpl_encode = _encoderImpl_encode impl . bw
+      , _encoderImpl_decode = fmap fw . _encoderImpl_decode impl
+      }
+
+-- | Lift an 'Iso' over the encoded format to an 'Iso' over encoders
+isoViaEncoded
+  :: Functor check
+  => Iso' e e'
+  -> Iso' (Encoder check parse d e) (Encoder check parse d e')
+isoViaEncoded i = withIso isoImpl $ \forward backward ->
+  iso
+    (Encoder . fmap forward . unEncoder)
+    (Encoder . fmap backward . unEncoder)
+  where
+    isoImpl = withIso i $ \forward backward -> iso (f backward forward) (f forward backward)
+    f bw fw impl = EncoderImpl
+      { _encoderImpl_encode = fw . _encoderImpl_encode impl
+      , _encoderImpl_decode = _encoderImpl_decode impl . bw
+      }
+
 wrappedEncoder :: (Wrapped a, Applicative check, Applicative parse) => Encoder check parse (Unwrapped a) a
 wrappedEncoder = isoEncoder $ from _Wrapped'
 
 unwrappedEncoder :: (Wrapped a, Applicative check, Applicative parse) => Encoder check parse a (Unwrapped a)
 unwrappedEncoder = isoEncoder _Wrapped'
 
+-- | Combine two encoders into a more complex one given isomorphisms for decoded/encoded formats
+pairEncoders
+  :: (Applicative check, Monad parse)
+  => Iso' (d1, d2) d
+  -> Iso' (e1, e2) e
+  -> (Encoder check parse d1 e1, Encoder check parse d2 e2)
+  -> Encoder check parse d e
+pairEncoders isoD isoE = view (isoViaDecoded isoD) . view (isoViaEncoded isoE) . uncurry bimap
```

Comment:
This would make more sense to me as something like
```haskell
pairEncoders isoD isoE (f, g) = isoEncoder isoE . bimap f g . isoEncoder (from isoD)
```
(not sure I got that exactly right)

Reply:
Indeed - fixed

Reply:
I'm no longer convinced `pairEncoders` is worth having though